### PR TITLE
cloud-games: add custom projects setting

### LINKED ISSUE #1652: Be able to add custom games to cloud games list
I'm making a multiplayer game and I'd like to keep a count of how many players are online at what time
**Describe the solution you'd like**
Be able to put custom games in the cloud games list

### PR DESCRIPTION
**Pull request by apple502j**
Resolves #1652

### Changes
Adds custom projects/studios setting to cloud-games addon, this time using table settings. There is also a button to add the project/studio from the selected tab.

Technical changes include `addon.popup.changeSettings` API and refactoring of button styles to share one CSS file.

### Reason for changes
Because I said I'll work on it, and it's been 1 year since my initial attempt (see #2871)

### CODE REVIEW

**Comment by Samq64** [comment]
I don't really like this:
image
Shouldn't it just always use the table setting and have Griffpatch's projects by default? And if the other option still fetches from a studio and you want to keep it, why not just make the studio Id customizable as well as having the table option?

**Comment by apple502j** [comment]
@Samq64 Project URL is just for user's convenience (ID works as well).

I don't want to overcomplicate settings page more than necessary, so the page won't lookup the API. I'll probably add a "project" button to popup, perhaps and add support for using studio URLs there.

**Comment by apple502j** [comment]
Scratch Addons popup says \"You can also click Add this studio below to add from the current tab.\"
Add this project button
New cloud games settings

File: popups/cloud-games/popup.js [hunk 1]
```diff
@@ -57,38 +161,51 @@ export default async ({ addon, msg, safeMsg }) => {
...
+          return;
+        }
+        throw e;
       }
-      const projects = await res.json();
       if (projects.length === 0) {
         this.error = "no-projects";
         return;
       }
-      // TODO: add currently opened game to projects array. Sort function should put it on top
```
Comment (Norbiros) [inline popups/cloud-games/popup.js:89]: Can you add this feature? (Check if online users includes current user name, if yes move it to top, and add visual effect for example add orange border)

**Comment by apple502j** [comment]
@RedGuy12 Fixed

File: popups/cloud-games/popup.js (hunk 1 above)
Comment (apple502j) [reply popups/cloud-games/popup.js:89]: Not for now

File: addons/cloud-games/addon.json [hunk 2]
```diff
@@ -15,7 +15,64 @@
...
+          "values": {
+            "url": "https://scratch.mit.edu/studios/539952/"
+          }
+        },
+        {
+          "name": "Massive Multiplayer Platformer",
+          "values": {
+            "url": "https://scratch.mit.edu/projects/612229554/"
+          }
+        },
```
Comment (WorldLanguages) [inline addons/cloud-games/addon.json:48]: Griffpatch took down this project :(

Comment (lisa-wolfgang) [reply addons/cloud-games/addon.json]: > MMO has been taken down again due to hackers
https://scratch.mit.edu/users/griffpatch

**Pull request merged**